homeassistant/components/rfxtrx/__init__.py

```diff
@@ -299,15 +298,12 @@ def get_new_device(event, config, device):
     return new_device
 
 
-def apply_received_command(event):
-    """Apply command from rfxtrx."""
-    device_id = slugify(event.device.id_string.lower())
-    # Check if entity exists or previously added automatically
-    if device_id not in RFX_DEVICES:
-        return
-
+def apply_received_command(event, entity):
```

Comment:
We shouldn't access the entity from outside. This is what we should use the dispatch helper to do. Register entity methods as signal targets.

Reply:
I know. But like you said, I'm taking this step på step.

Reply:
Ok.

Reply:
Actually this is not from outside anymore. It's just a helper from the entities reaction on dispatcher. So technically your comment is solved here already. Previously it did just as you said though.

That said, the function should be moved into the platforms that need it.

Reply:
We don't want a helper function that access entity methods from outside the entity. The signal target should be am entity method itself.

Reply:
I ended up resolving this. Moved into member functions of the individual platform.

PS. You don't happen to have a device do you? Could use some more manual test coverage on the next round of changes.

Reply:
No, sorry.